src/main/java/org/jenkinsci/plugins/mesos/MesosCleanupThread.java

```diff
@@ -0,0 +1,110 @@
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in
+ all copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ THE SOFTWARE.
+ */
+@Extension
+public class MesosCleanupThread extends AsyncPeriodicWork {
+
+    public MesosCleanupThread() {
+        super("Mesos pending deletion slave cleanup");
+    }
+
+    @Override
+    public long getRecurrencePeriod() {
+        return MIN * 5;
+    }
+
+    public static void invoke() {
+        getInstance().run();
+    }
+
+    private static MesosCleanupThread getInstance() {
+        return Jenkins.getInstance().getExtensionList(AsyncPeriodicWork.class).get(MesosCleanupThread.class);
+    }
+
+    @Override
+    protected void execute(TaskListener listener) {
+        final ImmutableList.Builder<ListenableFuture<?>> deletedNodesBuilder = ImmutableList.<ListenableFuture<?>>builder();
+        ListeningExecutorService executor = MoreExecutors.listeningDecorator(Computer.threadPoolForRemoting);
+        final ImmutableList.Builder<MesosComputer> computersToDeleteBuilder = ImmutableList.<MesosComputer>builder();
+
+        for (final Computer c : Jenkins.getInstance().getComputers()) {
+            if (MesosComputer.class.isInstance(c)) {
+                if (((MesosComputer) c).getNode().isPendingDelete()) {
+                    final MesosComputer comp = (MesosComputer) c;
+                    computersToDeleteBuilder.add(comp);
+                    logger.log(Level.INFO, "Marked " + comp.getName() + " for deletion");
+                    ListenableFuture<?> f = executor.submit(new Runnable() {
+                        public void run() {
+                            logger.log(Level.INFO, "Deleting pending node " + comp.getName());
+                            try {
+                                comp.getNode().terminate();
+                            } catch (RuntimeException e) {
+                                logger.log(Level.WARNING, "Failed to disconnect and delete " + c.getName() + ": " + e.getMessage());
+                                throw e;
+                            }
+                        }
+                    });
+                    deletedNodesBuilder.add(f);
+                } else {
+                    logger.log(Level.INFO, c.getName() + " is not pending deletion");
```

Comment:
Should this be a DEBUG as it is the normal state? This plugin already produces a lot of INFO level output.